### PR TITLE
Add STM32F413 support

### PR DESCRIPTION
**Pull request by vaussard**
This adds support for STM32F413xH, as well as the Nucleo STM32F413 board.

### CODE REVIEW

File: arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.stm32f413xx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+if SOC_STM32F413XH
+
+config SOC
+	string
+	default stm32f413xx
+
+config NUM_IRQS
+	int
+	default 101
```
Comment (erwango) [inline arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.stm32f413xx]: 102

**Review by galak — CHANGES_REQUESTED** [review]
should add the board to  scripts/sanity_chk/arches/arm.ini

File: arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.stm32f413xx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Kconfig - ST STM32F413XH MCU configuration options
+#
+# Copyright (c) 2017 Florian Vaussard, HEIG-VD
```
Comment (erwango) [inline arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.stm32f413xx]: We're not using names in copyright, but I don't know if this is a real rule.
I know some do, but some other have been rejected.
@galak, you might know if there is a strict rule?

**Review by erwango — CHANGES_REQUESTED** [review]

File: arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.stm32f413xx (hunk 2 above)
Comment (galak) [inline arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.stm32f413xx]: I don't think there is a strict rule here, usually goes to company choice

**Review by galak — CHANGES_REQUESTED** [review]
Missing board docs

File: arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.stm32f413xx (hunk 1 above)
Comment (vaussard) [inline arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.stm32f413xx]: Fixed, thank you!

**Comment by vaussard** [comment]
I added the board to scripts/sanity_chk/arches/arm.ini and added the board doc. Thank you

**Review by dbkinder — APPROVED** [review]
LGTM

**Review by galak — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**